.gitlab/deps_fetch.yml

```diff
@@ -8,11 +8,11 @@
   needs: []
   script:
     - inv -e deps --verbose
-    - cd $CI_PROJECT_DIR/vendor && tar czf $CI_PROJECT_DIR/vendor.tar.gz .
+    - cd $GOPATH/pkg/mod/ && tar czf $CI_PROJECT_DIR/modcache.tar.gz .
```

Comment:
We compress `$GOPATH/pkg/mod` but extract to `/gomodcache`. It might be safe to move the `mod` directory around, but I'm not 100% sure it is (eg: if it contains links, they could break). Is there any reason not to use always the same path (eg: `/gomodcache`)?

Reply:
No technical reason against it; just that it would mean a bigger change (on every agent flavor build task) and this seems to work fine. We already had that problem with vendoring: the retrieve jobs copy from `/go/<path to datadog-agent repo>/vendor` to `/.omnibus/src/<omnibus project name>/vendor` so I would rather do this in a separate PR if we think it's needed.